### PR TITLE
Fix pulling images from Docker Hub

### PR DESCRIPTION
**Pull request by abvgedeika**

### CODE REVIEW

File: lib/services/dapi/createDapi.js [hunk 1]
```diff
@@ -23,7 +23,12 @@ async function createDapi(opts) {
   const authorizationToken = await getAwsEcrAuthorizationToken(options.getAwsOptions());
```
Comment (shumkov) [inline lib/services/dapi/createDapi.js]: You don't need to fetch token if you are not using it.

File: lib/services/dapi/Dapi.js [hunk 2]
```diff
@@ -76,7 +76,7 @@ class Dapi extends NodeJsService {
           starting = false;
         }
       } catch (error) {
-        if (error.message.indexOf("Merkle root from the diff doesn't match calculated merkle root after diff is applied") > -1) {
+        if (error.message.indexOf("DAPI RPC error: getBestBlockHeight: TypeError: Cannot read property 'length' of undefined") > -1 || error.message.indexOf("DAPI RPC error: getBestBlockHeight: Error: No MNs in list. Can't connect to the network.") > -1) {
```
Comment (shumkov) [inline lib/services/dapi/Dapi.js]: Shall we fix it in a separate PR?

Comment (shumkov) [reply lib/services/dapi/Dapi.js]: By the way, why do we change it? :)

File: lib/services/dapi/createDapi.js [hunk 3]
```diff
@@ -23,7 +23,12 @@ async function createDapi(opts) {
...
 
   const imageName = options.getContainerImageName();
-  const image = new Image(imageName, authorizationToken);
+  let image;
+  if (imageName.indexOf('amazonaws.com') > 0) {
+    image = new Image(imageName, authorizationToken);
+  } else {
+    image = new Image(imageName, null);
```
Comment (shumkov) [inline lib/services/dapi/createDapi.js]: Please make the second argument as an optional.

File: lib/services/dapi/createDapi.js (hunk 1 above)
Comment (abvgedeika) [reply lib/services/dapi/createDapi.js]: changed

File: lib/services/dapi/Dapi.js (hunk 2 above)
Comment (abvgedeika) [reply lib/services/dapi/Dapi.js]: I don't know) @Cofresi @antouhou  do you have any ideas with this new behavior?

Comment (shumkov) [reply lib/services/dapi/Dapi.js]: Could we fix it separately? It's out of scope.

File: lib/docker/Image.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ class Image {
    * @param {String} [image]
    * @param {Object} authorizationToken
```
Comment (shumkov) [inline lib/docker/Image.js]: Please mark it as optional:
```js
/**
   * @param {String} image
   * @param {Object} [authorizationToken]
```

File: lib/services/dapi/createDapi.js [hunk 5]
```diff
@@ -20,10 +20,14 @@ async function createDapi(opts) {
   const { name: networkName, driver } = options.getContainerNetworkOptions();
   const network = new Network(networkName, driver);
 
-  const authorizationToken = await getAwsEcrAuthorizationToken(options.getAwsOptions());
-
   const imageName = options.getContainerImageName();
-  const image = new Image(imageName, authorizationToken);
+  let image;
```
Comment (shumkov) [inline lib/services/dapi/createDapi.js]:
```suggestion
let authorizationToken;
if (imageName. includes('amazonaws.com')) {
    authorizationToken = await getAwsEcrAuthorizationToken(options.getAwsOptions());
}

const image = new Image(imageName, authorizationToken);
```

File: lib/services/dapi/Dapi.js (hunk 2 above)
Comment (abvgedeika) [reply lib/services/dapi/Dapi.js]: without this fix dapi tests failed. or you suggest to disable them at all?

File: lib/services/dapi/createDapi.js (hunk 5 above)
Comment (abvgedeika) [reply lib/services/dapi/createDapi.js]: modified

File: lib/docker/Image.js (hunk 4 above)
Comment (abvgedeika) [reply lib/docker/Image.js]: added

**Pull request merged**